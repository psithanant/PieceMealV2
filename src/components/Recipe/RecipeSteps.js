import React, { Component } from 'react';
import { ListView, Switch } from 'react-native';
import { Container, Content, View, Card, CardItem, Text } from 'native-base';

export default class RecipeSteps extends Component {

  state = {
    toggled: false
  }

  toggleSwitch() {
      this.setState({ toggled: !this.state.toggled });
  }

  loadDataSource() {
       const ds = new ListView.DataSource({
           rowHasChanged: (r1, r2) => r1 !== r2
       });

       this.dataSource = ds.cloneWithRows(this.props.instructions);
   }

  renderRow(rowData) {
    return (
        <CardItem>
            <Switch value={this.state.toggled} onValueChange={() => { this.toggleSwitch(); }} />
            {/* <Switch value={this.state.toggled} onValueChange={() => { this.value = !this.value; }} /> */}
            <Text
              style={{ fontStyle: 'italic',
                       alignSelf: 'center',
                       paddingLeft: 10,
                       fontFamily: 'Futura',
                       color: '#373737'
                    }}
            >
              { rowData.instructions }
            </Text>
        </CardItem>

        // <Card >
        //     <CardItem>
        //         <Text
        //           style={{ fontStyle: 'italic',
        //                    alignSelf: 'center',
        //                    fontFamily: 'Futura',
        //                    fontSize: 14,
        //                    color: '#373737',
        //                    flex: 3 }}
        //         >
        //           { rowData.instructions }
        //         </Text>
        //     </CardItem>
        // </Card>
    );
  }

  render() {
    this.loadDataSource();
      return (
        <Content>
          <View style={{ marginBottom: 35 }}>
            <ListView
              enableEmptySections
              dataSource={this.dataSource}
              renderRow={this.renderRow.bind(this)}
            />
          </View>
        </Content>
      );
    }
}
