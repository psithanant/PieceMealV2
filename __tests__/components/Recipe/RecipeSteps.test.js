import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RecipeSteps from '../../../src/components/Recipe/RecipeSteps';

describe('tests RecipeSteps', () => {
  it('renders RecipeSteps component', () => {
  const component = shallow(
    <RecipeSteps
      instructions={[]}
    />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
