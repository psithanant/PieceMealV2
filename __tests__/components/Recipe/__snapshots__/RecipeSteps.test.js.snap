// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests RecipeSteps renders RecipeSteps component 1`] = `
<Styled(Content)>
  <Styled(ViewNB)
    style={
      Object {
        "marginBottom": 35,
      }
    }
  >
    <ListViewMock
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    />
  </Styled(ViewNB)>
</Styled(Content)>
`;
